### PR TITLE
test: Add test for ImagesTable.js

### PR DESCRIPTION
**Pull request by henrywang**
1. Add images table render test
2. Update composes prop type to array and change composes default value to empty array instead of empty object in testUtils.renderWithReduxRouter

API calling part will be in another PR (which needs mock)

### CODE REVIEW

File: src/SmartComponents/ImagesTable/ImagesTable.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ function mapDispatchToProps(dispatch) {
 }
 
 ImagesTable.propTypes = {
-    composes: PropTypes.object,
+    composes: PropTypes.array,
```
Comment (Gundersanne) [inline src/SmartComponents/ImagesTable/ImagesTable.js:105]: Why change this to an array?

**Review by Gundersanne — COMMENTED** [review]
I tried to do the same thing yesterday here https://github.com/osbuild/image-builder-frontend/pull/45/commits/09274f7234adddeb589aee14eb476187360dc30c but with Jenn's ImageBuildStatus included. I'm not sure what you would prefer, do you want to merge this first?

Also `composes` is an object with uuids as key.
```

const store = {
    composes: {
        'c1cfa347-4c37-49b5-8e73-6aa1d1746cfa': {
            status: 'building',
            distribution: 'fedora-31',
            architecture: 'x86_64',
            image_type: 'qcow2'
        },
        '61b0effa-c901-4ee5-86b9-2010b47f1b22': {
            status: 'uploading',
            distribution: 'fedora-31',
            architecture: 'x86_64',
            image_type: 'qcow2'
        },
        '551de6f6-1533-4b46-a69f-7924051f9bc6': {
            status: 'success',
            distribution: 'fedora-31',
            architecture: 'x86_64',
            image_type: 'qcow2'
        }
    }
};

```



Do y

File: src/test/SmartComponents/ImagesTable/ImagesTable.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { renderWithReduxRouter } from '../../testUtils';
+
+import ImagesTable from '../../../SmartComponents/ImagesTable/ImagesTable';
+
+describe('ImagesTable Page', () => {
```
Comment (atodorov) [inline src/test/SmartComponents/ImagesTable/ImagesTable.test.js:7]: from what I understand this isn't a stand-alone page

File: src/test/SmartComponents/ImagesTable/ImagesTable.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        };
+        renderWithReduxRouter(<ImagesTable />, mockedStore);
+    });
+    test('renders images table', () => {
+        // check images table
+        screen.getByTestId('images-table');
+
+        // check props render
+        screen.getByText(/fedora-33/i,);
+        screen.getByText(/qcow2/i,);
```
Comment (atodorov) [inline src/test/SmartComponents/ImagesTable/ImagesTable.test.js:25]: is there any reason to have the search not case sensitive ?

File: src/test/SmartComponents/ImagesTable/ImagesTable.test.js (hunk 2 above)
Comment (Gundersanne) [reply src/test/SmartComponents/ImagesTable/ImagesTable.test.js:7]: I think it's fine to test components on their own no?

**Comment by atodorov** [comment]
@henrywang can you also rebase to latest master. I think codecov is doing a poor job updating the results for the master branch (at least not immediately) but statuses reported on pull requests seem to be correct. We should be at 51% before your new test.

File: src/test/SmartComponents/ImagesTable/ImagesTable.test.js (hunk 2 above)
Comment (atodorov) [reply src/test/SmartComponents/ImagesTable/ImagesTable.test.js:7]: It is, just the wording is misleading b/c this is a component, not a stand alone page.

**Comment by henrywang** [comment]
> I tried to do the same thing yesterday here [09274f7](https://github.com/osbuild/image-builder-frontend/commit/09274f7234adddeb589aee14eb476187360dc30c) on PR #45 but with Jenn's ImageBuildStatus included. I'm not sure what you would prefer, do you want to merge this first?
> 
> Also `composes` is an object with uuids as key.
> 
> ```
> 
> const store = {
>     composes: {
>         'c1cfa347-4c37-49b5-8e73-6aa1d1746cfa': {
>             status: 'building',
>             distribution: 'fedora-31',
>             architecture: 'x86_64',
>             image_type: 'qcow2'
>         },
>         '61b0effa-c901-4ee5-86b9-2010b47f1b22': {
>             status: 'uploading',
>             distribution: 'fedora-31',
>             architecture: 'x86_64',
>             image_type: 'qcow2'
>         },
>         '551de6f6-1533-4b46-a69f-7924051f9bc6': {
>             status: 'success',
>             distribution: 'fedora-31',
>             architecture: 'x86_64',
>             image_type: 'qcow2'
>         }
>     }
> };
> ```

@Gundersanne, oh, that's fine. let's land PR #45 first.

**Comment by henrywang** [comment]
> @henrywang can you also rebase to latest master. I think codecov is doing a poor job updating the results for the master branch (at least not immediately) but statuses reported on pull requests seem to be correct. We should be at 51% before your new test.

@atodorov, ack. Thanks.

**Comment by Gundersanne** [comment]
I reopened this PR in #54 against main.

**Pull request closed without merging**